src/plot_api/edit_types.js

```diff
@@ -0,0 +1,84 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var isPlainObject = require('../lib/is_plain_object');
+
+module.exports = {
```

Comment:
Actually, turns out putting these in helpers makes new circular references and breaks plotting entirely.

- plots/plots, plot_api/plot_schema, plot_api/helpers
- plots/cartesian/axes, components/titles/index, plotly, components/modebar/index, components/modebar/manage, components/modebar/buttons, plots/plots, plot_api/plot_schema, plot_api/helpers

The real culprit here seems to be `plots/plots.js`, that ancient beast is coupled to everything under the sun (and indeed is part of 3 other loops too)... but do you mind if I leave `edit_types.js` as a separate file for now?